src/btree/bt_rebalance.c

```diff
@@ -0,0 +1,485 @@
+/*-
+ * Copyright (c) 2014-2016 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * Shared rebalance information.
+ */
+typedef struct {
+	WT_REF **leaf;				/* List of leaf pages */
+	size_t	 leaf_next;			/* Next entry */
+	size_t   leaf_allocated;		/* Allocated bytes */
+
+	WT_ADDR *fl;				/* List of objects to free */
+	size_t	 fl_next;			/* Next entry */
+	size_t   fl_allocated;			/* Allocated bytes */
+
+	WT_PAGE *root;				/* Created root page */
+
+	uint8_t	 type;				/* Internal page type */
+
+#define	WT_REBALANCE_PROGRESS_INTERVAL	100
+	uint64_t progress;			/* Progress counter */
+
+	WT_ITEM *tmp1;				/* Temporary buffers */
+	WT_ITEM *tmp2;
+} WT_RSTUFF;
```

Comment:
I know it's local to the file and matches salvage/verify, but I don't like the name. Could we call it ``WT_REBALANCE_STUFF``? It's not used in line-length critical places.